examples/interfaces/10.backup.cpp

```diff
@@ -0,0 +1,175 @@
+ *  for the Firebird Open Source RDBMS project.
+ *
+ *  Copyright (c) 2016 Alexander Peshkoff <peshkoff@mail.ru>
+ *  and all contributors signed below.
+ *
+ *  All Rights Reserved.
+ *  Contributor(s): ______________________________________.
+ *
+ */
+
+#include "ifaceExamples.h"
+
+static IMaster* master = fb_get_master_interface();
+
+// print information, returned by isc_svc_query() call
+
+bool printLine(const unsigned char*& p)
+{
+	const ISC_USHORT length = (ISC_USHORT) isc_vax_integer((char*)p, sizeof(ISC_USHORT));
+	p += sizeof(ISC_USHORT);
+	if (length > 0)
+		printf("%*.*s\n", length, length, p);
+	p += length;
+	return length > 0;
+}
+
+bool printInfo(const unsigned char* p, size_t pSize)
+{
+	bool ret = false;
+	bool ignoreTruncation = false;
+	const unsigned char* const end = p + pSize;
+
+	while (p < end && *p != isc_info_end)
+	{
+		switch (*p++)
+		{
+		case isc_info_svc_line:
+			ret = printLine(p);
+			break;
+
+		case isc_info_truncated:
+			if (!ignoreTruncation)
+			{
+				printf("\n<<< truncated >>>\n");
+			}
+			fflush(stdout);
+			ret = true;
+			break;
+
+		case isc_info_svc_timeout:
+		case isc_info_data_not_ready:
+			ret = true;
+			break;
+
+		default:
+			printf("Unknown item 0x%x in received buffer\n", p[-1]);
+		}
+	}
+
+	fflush(stdout);
+	return ret;
+}
+
+
+int main()
+{
+	int rc = 0;
+
+	// Status wrapper
+	ThrowStatusWrapper status(master->getStatus());
+
+	// Declare pointers to required interfaces
+	IProvider* prov = master->getDispatcher();
+	IUtil* utl = master->getUtilInterface();
+	IService* svc = NULL;
+	IXpbBuilder* spb1 = NULL;
+	IXpbBuilder* spb2 = NULL;
+
+	try {
+		printf("** Attaching to service manager...\n");
+
+		// Prepare SPB to attach to service manager
+		spb1 = utl->getXpbBuilder(&status, IXpbBuilder::SPB_ATTACH, NULL, 0);
+		spb1->insertString(&status, isc_spb_user_name, "sysdba");
+		spb1->insertString(&status, isc_spb_password, "masterkey");
+		// In case when your program is expected to be used on a server
+		// with multiple security database it's very good idea to specify
+		// DB expected to be used by services (you anyway need separate
+		// services connectons for databases with different user list location).
+		spb1->insertString(&status, isc_spb_expected_db, "fbtests.fdb");
+
+		// Attach to service manager
+		svc = prov->attachServiceManager(&status, "service_mgr", spb1->getBufferLength(&status),
+			spb1->getBuffer(&status));
+
+		// In the simplest case sendItems parameter may be NULL
+		// Building receiveItems is mostly trivial
+		const unsigned char receiveItems[] = {isc_info_svc_server_version};
+
+		// Output buffer
+		unsigned char results[1024];
+
+		printf("** Demo of making backup using service manager...\n");
+
+		// Build service start SPB
+		spb2 = utl->getXpbBuilder(&status, IXpbBuilder::SPB_START, NULL, 0);
+		spb2->insertTag(&status, isc_action_svc_backup);
+		spb2->insertString(&status, isc_spb_dbname, "fbtests.fdb");
```

Comment:
@aafemt I think it will be correct to use employee in backup and what else you wish at restore. employee_copy.fdb or anything else you wish. It's good idea when samples run out of the box.